### PR TITLE
Unsafe YAML dumping

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Adds an `AnsibleUnsafeDumper` which dumps unsafe strings with the `!unsafe` tag.

The second commit adds a `!unsafe-binary` tag so that unsafe binary strings can also be dumped and survive a round-trip with `ansible-inventory --list --yaml`.

Ref: #82999.

##### ISSUE TYPE
- Bugfix Pull Request
- Feature Pull Request

### CODE REVIEW

**Comment by nitzmahone** [comment]
I keep half-finishing a followup on this, getting distracted, then crashing my browser before I get back to it... :neutral_face: 

TL;DR, we probably don't want to handle it this way...

Data tagging's switch to "everything is untrusted as template by default" makes this mostly unnecessary. There's a knob on our YAML deserializer to tag all values as trusted for templating or not (defaults to true on anything loaded by our dataloader), and if you want more granular control, you can either use the existing `!unsafe` tag to let the doc de-trust individual values (strips `TrustedAsTemplate` and adds `NotATemplate` to avoid spurious warnings), or you can muck with individual values and tag/untag any way you want in code after the document is loaded. We can already preserve `!unsafe` when it's explicitly marked, and we can already make that work for scalars/mappings/lists without a new tag.

Custom YAML tags are problematic for a number of reasons- the main one being that they're intended for type mapping, not metadata. It's possible to abuse them for metadata with a custom tag scheme, but it gets ugly *really* fast. There's also the problem of providing granular controls over serialization/deserialization around which tags should be emitted/recognized and when (e.g., is this document being serialized to be read by Ansible or some other YAML thing that will barf on custom tags?). For those reasons (and several others), we've been trying to avoid creating any new YAML tags, and minimize the need for the ones we already have. Long story short: we're leaning a lot harder toward "JSON is our only supported full-fidelity round-trippable serialization format".

**Comment by felixfontein** [comment]
In that case I'm closing this PR.
